### ISSUE TITLE
docs.rs should ignore .cargo/config

### ISSUE THREAD

**Issue by jamwaffles** [rust-lang/docs.rs#647]
**Crate name:** [`sh1106`](https://crates.io/crates/sh1106)
**Build failure link:** https://docs.rs/crate/sh1106/0.3.0/builds/231814

**Additional details:**

I'm building a no_std crate targeting `thumbv7m-none-eabi`, etc. To get docs.rs builds working again, I added the following to my `Cargo.toml`:

```toml
[package.metadata.docs.rs]
default-target = "x86_64-unknown-linux-gnu"
```

Which doesn't seem to have an effect, but _only on docs.rs_. If I build it locally using the `docs.rs` repo and the following command, it builds fine.

```bash
docker-compose run -v "$(realpath ~/Repositories/sh1106)":/build web build crate --local /build
```

I do have a `.cargo/config` with the following at the end which may be causing issues?

```toml
[build]
target = "thumbv7m-none-eabi"
```

**Comment by jyn514** [comment]
Hmm I thought we fixed this in https://github.com/rust-lang/docs.rs/pull/633 ... In the meantime you can workaround it by removing .cargo/config.

Maybe the issue is that since we don't pass `--target` to cargo it uses the value in `.cargo/config`? In that case setting`default-target` to anything other than `x86_64-unknown-linux-gnu` should also fix the build error.

I wish there were a way to get cargo to tell us what target in plans to build ... Maybe `cargo metadata` would have that info.

**Comment by jyn514** [comment]
The fact that it works locally but not in production is also very confusing, I have no idea what could cause that.

**Comment by jyn514** [comment]
> The fact that it works locally but not in production is also very confusing, I have no idea what could cause that.

Oh I do have an idea - if you'd set `target = "thumbv7m-none-eabi"` earlier and run a build it would have cached the toolchain, so it would still be installed even when docs.rs didn't install it later.

**Comment by jamwaffles** [comment]
That makes sense! I removed every image and volume I had locally and reran the build which reproduces the error I see on docs.rs.

I've changed the docs.rs config in `Cargo.toml` to this now:

```toml
[package.metadata.docs.rs]
targets = [ "thumbv7m-none-eabi", "thumbv7em-none-eabihf" ]
```

Is that the best way to go? I realise it's a very recent addition but if it works on docs.rs then it works great for me :grin:.

**Comment by jyn514** [comment]
Yes, that should work. Thanks for working with us, I'm glad the new feature is helpful!

**Comment by jyn514** [comment]
Fixing the root problem should be basically the same steps as https://github.com/rust-lang/docs.rs/issues/555#issuecomment-571950211, rustwide should remove `.cargo/config` before starting the build.

**Comment by jamwaffles** [comment]
Thanks for the suggestions! My builds are working again. Please close this issue if you think it should be :)

**Comment by jyn514** [comment]
I'll open a new issue for ignoring `.cargo/config` so you don't get spammed with notifications.
